packages/meteor/url_common.js

```diff
@@ -43,8 +43,9 @@ Meteor.absoluteUrl = function (path, options) {
       !/http:\/\/127\.0\.0\.1[:\/]/.test(url)) // or 127.0.0.1
     url = url.replace(/^http:/, 'https:');
 
-  if (options.replaceLocalhost)
-    url = url.replace(/^http:\/\/localhost([:\/].*)/, 'http://127.0.0.1$1');
+  var localhostRegex = /^http:\/\/localhost([:\/].*)/;
+  if (localhostRegex.test(url) || options.replaceLocalhost)
```

Comment:
Any reason to replace localhost by default if the URL contains the localhost regex, instead of respecting the flag? Wouldn't it be a `&&`?

Reply:
This API had to be changed because of how node server handles fetching files inside of it. We are planning on documenting what changed (this will be probably a breaking change)

Reply:
We updated the tests for this function to match what is the behavior now. Only thing missing is docs

Reply:
- If `url` **matches** against `locationRegex` and `options.replaceLocalhost` is **true**, this line will perform a string replacement in `url`.
- If `url` **matches** against `locationRegex` and `options.replaceLocalhost` is **false**, this line will perform a string replacement in `url`.
- If `url` **doesn't match** against `locationRegex` and `options.replaceLocalhost` is **true**, this line will do a trivial assignment (`url = url;`) because `url.replace()` will not find a matching substring; it could be just skipped.
- If `url` **doesn't match** against `locationRegex` and `options.replaceLocalhost` is **false**, this line will be skipped.

The `if` condition is unnecessary and just adds a regexp lookup (`localhostRegex.test(url)`) that `String.prototype.replace` already does.